sql/core.go

```diff
@@ -370,6 +370,11 @@ type SystemVariable struct {
 	// block.  NotifyChanged is not called when a new system variable is
 	// registered.
 	NotifyChanged func(SystemVariableScope, SystemVarValue)
+	// ValueFunction defines an optional function that is executed to provide
+	// the value of this system variable whenever it is requested. System variables
+	// that provide a ValueFunction should also set Dynamic to false, since they
+	// cannot be assigned a value and will return a read-only error if tried.
+	ValueFunction func() interface{}
```

Comment:
I tried out an `error` return param, too, but was kinda on the fence about it, since neither function needed it yet and I figured we could add it easily later, but happy to add it now if you feel strongly about it.